src/mbgl/map/map.cpp

```diff
@@ -67,6 +67,7 @@ class Map::Impl : public style::Observer {
 
     std::string styleURL;
     std::string styleJSON;
+    bool styleMutated = false;
```

Comment:
Can we move this to the `Style` object and have it keep track of whether it was mutated itself?